php/media/class-filter.php

```diff
@@ -371,6 +376,52 @@ public function filter_out_local( $content ) {
 		return $this->filter_video_shortcodes( $content );
 	}
 
+	/**
+	 * Apply loading=lazy to <img> if necessary.
+	 *
+	 * @param string $new_tag
+	 * @param bool   $is_admin
+	 *
+	 * @return string
+	 */
+	protected function add_lazy_loading_attribute( $new_tag, $is_admin = false ) {
+		if (
+			apply_filters( 'wp_lazy_loading_enabled', true ) &&
+			false === strpos( $new_tag, 'loading="lazy"' ) &&
+			$is_admin
+		) {
+			$new_tag = str_replace( '/>', ' loading="lazy" />', $new_tag );
+		}
+
+		return $new_tag;
+	}
+
+	/**
+	 * Checks whether $cloudinary_url has f_mp4 which converts gifs to mp4s.
+	 *
+	 * @param string $cloudinary_url
+	 * @param int    $attachment_id
+	 *
+	 * @return string|null
+	 */
+	protected function check_fmp4_presence( $cloudinary_url, $attachment_id ) {
+		$return = null;
+
+		if ( Utils::ends_with( $cloudinary_url, '.gif' ) ) {
```

Comment:
@dugajean @pereirinha  There's no reason to search for end like this since we have the ID we can simply check the mime_type:
```suggestion
		if ( 'image/gif' === get_post_mime_type( $attachment_id ) ) {
```